crates/bevy_math/src/curves/fixed.rs

```diff
@@ -0,0 +1,123 @@
+use crate::{curves::Curve, interpolation::Lerp};
+
+// TODO: impl Serialize, Deserialize
+/// Curve with evenly spaced keyframes, in another words a curve with a fixed frame rate.
+///
+/// This curve maintains the faster sampling rate over a wide range of frame rates, because
+/// it doesn't rely on keyframe cursor. But as a downside it will have the bigger memory foot print;
+#[derive(Default, Debug)]
+pub struct CurveFixed<T> {
+    // ? NOTE: Has I learned from benches casting to f32 is quite expensive
+    // ? so frame rate and offset values must be stored as f32
+    frame_rate: f32,
+    /// Negative number of frames before the curve starts
+    negative_offset: f32,
```

Comment:
negative offset seems very strange to me while reading the code, could it be changed so that's it's a normal offset?

Reply:
It's an internal value meant, I do keep as negative value to improve perf, the public api handles the offset as it should be